src/Storages/RabbitMQ/WriteBufferToRabbitMQProducer.cpp

```diff
@@ -322,7 +275,7 @@ void WriteBufferToRabbitMQProducer::writingFunc()
 
         if (wait_num.load() && delivery_record.empty() && payloads.empty() && returned.empty())
             wait_all = false;
-        else if ((!producer_channel->usable() && event_handler->connectionRunning()) || (!event_handler->connectionRunning() && setupConnection(true)))
+        else if ((!producer_channel->usable() && connection.isConnected()) || (!connection.isConnected() && connection.reconnect()))
```

Comment:
It would be better to combine `!connection.isConnected() && connection.reconnect()` in one method under one mutex locking.